src/utils/dispatcher.js

```diff
@@ -2,16 +2,16 @@ import client from '../client';
 import { prefix, validCmds } from '../constants';
 
 export const parseMessage = (msg) => {
-  msg = msg.toString();
-  if (msg.substring(0, 2) === prefix) {
-    const args = msg.substring(2).split(' ');
+  const msgText = msg.toString();
+  if (msgText.substring(0, 2) === prefix) {
+    const args = msgText.substring(2).split(' ');
     const cmd = args[0];
     if (validCmds.includes(cmd)) {
-      client.emit('command', cmd, args.splice(1));
+      client.emit('command', cmd, args.splice(1), msg);
```

Comment:
Yeah see my comment above about mapping a cmd to its respective function in an enum.

Reply:
Responded above, resolving